### ISSUE TITLE
[Feature]: Is there a way to add a footer row or a filter row in datagrid

### ISSUE THREAD

**Issue by it950** [microsoft/fluentui#32257]
### Library

React Components / v9 (@fluentui/react-components)

### Describe the feature that you would like added

1, How to add a footer to DataGrid, the footer show the sum field and sticky to the bottom of DataGrid, like DetailList v8           onRenderDetailsFooter property;
2, How to add a sticky filter row under the DataGridHeader, and user can filter the data in DataGrid

### Validations

- [X] Check that there isn't already an issue that requests the same feature to avoid creating a duplicate.

### Priority

None

**Comment by miroslavstastny** [comment]
This is currently not possible, tracking as a feature request

**Comment by DigitalFlow** [comment]
I would not say it's not possible.

I achieved this by adding a column with a specific ID and a custom sort function that puts this column always last / first depending on the sort direction.

Values have to be calculated by code then when rendering.
We even managed to implement collapsible rows with children like this.

**Comment by it950** [comment]
The footer is very user-friendly and the user can view the summary of the dataGrid row directly, in v8, this is very easy, the following pictue is DetailList footer in fluentui v8:
![image](https://github.com/user-attachments/assets/47c634d3-33a0-4993-ac07-b8b8b22d8057)

**Comment by DigitalFlow** [comment]
Yes you're right, but this is super easy to implement in datagrid as well.

You can just add an extra item to your "items" collection with a marker like an empty guid and in your createTableColumn functions you can check for this is and if you find it, don't render the columns value but instead access the whole items collection and aggregate it.

It's really not much effort.
